content/docs/user-guide/assets/pipeline/source-assets.md

```diff
@@ -0,0 +1,49 @@
+
+If you have many source assets to process, you might find using the Texture Settings and Scene Settings tools time consuming. The `.assetinfo` sidecar file is in JSON format and can be generated and modified with Python. It is possible, for example, to script the export process from a DCC application to automatically generate a `.assetinfo` file, or run a batch Python script over a source asset repository to add or modify options in the `.assetinfo` files. Many assets in O3DE are JSON formatted: `.material`, `.prefab`, and `.physicsmaterial` files, for example, are all in JSON format and can be automatically generated and modified with Python.
+
+Python can also be used to customize asset processing. Python scripts can be run during asset processing to modify how product assets are produced. **Python Asset Builder** provides a way to create custom asset processing jobs. For more information, refer to [Process custom assets with Python Asset Builder](./builder).
+
+With JSON and Python scripting, it's possible to customize and automate the entire asset pipeline, including integrating new tools and asset types.
+
+## Asset reprocessing
+
+Source asset files are tracked by timestamp, file hash, and Asset Builder fingerprint. To ensure the product assets for your project are always up to date, Asset Processor automatically schedules asset processing jobs in the following scenarios:
+
+### Changes to the source asset
+
+If a source asset is modified, and the timestamp of a source asset or its `.assetinfo` doesn't match the timestamp information of the last process job, Asset Processor adds a job to reprocess the source asset.
+
+{{< note >}}
+Asset Processor has an option to use file hashes instead of timestamps to detect file changes. Using hashes to detect file changes can significantly increase the analysis time for source assets, though. 
+{{< /note >}}
+
+### Changes to job dependencies
+
+Some assets have [job dependencies](asset-dependencies-and-identifiers). Changes to a job dependency trigger Asset Processor to schedule a job to reprocess source assets with that dependency. For example, materials are dependent on shaders. If a shader is modified and reprocessed, then all the materials that reference that shader are automatically reprocessed as well.
+
+### Changes to the Asset Builder
+
+Asset Builders use an engineer specified *fingerprint* to identify the version of the Asset Builder. The Asset Builder fingerprint is stored in the information about the job that produced the product asset. If an Asset Builder for a particular asset type has been updated and its fingerprint has been modified, it won't match the fingerprint of the jobs that processed the source assets supported by the Asset Builder. This triggers all source assets that use the Asset Builder to be reprocessed.

```

Comment:
"Engineer-specified" sounds a little too... personal? Consider rephrasing to something like the following suggestion.
```suggestion
Asset Builders use a *fingerprint* defined in their source code to identify the version of the Asset Builder. The Asset Builder fingerprint is stored in the information about the job that produced the product asset. If an Asset Builder for a particular asset type has been updated and its fingerprint has been modified, it won't match the fingerprint of the jobs that processed the source assets supported by the Asset Builder. This triggers all source assets that use the Asset Builder to be reprocessed.

```